syntax = "proto3";

package moose;

/* The Executor gRPC service was previously used by a gRPC runtime implementation,
 * which has since been deprecated. It can be found in
 * pymoose/deprecated/choreography/grpc.py. */

service Executor {
    rpc RunComputation(RunComputationRequest) returns(RunComputationResponse) {}
    rpc GetPublicKey(GetPublicKeyRequest) returns(GetPublicKeyResponse) {}
}

message RunComputationRequest {
    bytes computation = 1;
    string placement = 2;
    int64 session_id = 3;
    map<string, HostInfo> placement_instantiation = 4;
}

message HostInfo {
    string endpoint = 1;
    bytes public_key = 2;
}

message RunComputationResponse {}

message GetPublicKeyRequest {}

message GetPublicKeyResponse {
    bytes value = 1;
}
